tools/rdm/TestDefinitions.py

```diff
@@ -651,20 +661,39 @@ def GetFailed(self):
     self.SendGet()
 
 
-class SetDeviceInfo(ResponderTestFixture, DeviceInfoTest):
-  """SET device info."""
+class SetDeviceInfo(ResponderTestFixture,
+                    DeviceInfoTest):
+  """Attempt to SET device info with no data."""
   CATEGORY = TestCategory.ERROR_CONDITIONS
 
   def Test(self):
-    self.AddExpectedResults(TestMixins.UnsupportedSetNacks(self.pid))
-    self.SendRawSet(ROOT_DEVICE, self.pid)
+    # Unknown PID shouldn't be valid for a mandatory PID
+    self.AddExpectedResults(
+      self.NackSetResult(RDMNack.NR_UNSUPPORTED_COMMAND_CLASS)
```

Comment:
Can we compromise on a warning or advisory? I appreciate what you're saying, but you may as well only have one NACK code if you're going to ignore obvious precedence/appropriateness. Some similar stuff came up at the Plugfest too.. Aside from the word message being a poor choice, the standard says "The responder cannot comply with request because the message is not implemented in responder.". Assuming message=PID, it's not the right choice of NACK code on a mandatory PID. Also it would be less confusing as a controller writer if they use the more appropriate NACK case.